### PR TITLE
[H53] CP-LOSS-WEIGHT — surface-pressure channel up-weighting to attack test_SP binding gate

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**test_SP is the binding gate that has resisted 7+ Wave 31 experiments.** Every run since Wave 31 began lands test_SP at 3.7–3.9%, breaching our baseline floor of 3.577%. The vol_p pathway has 7 confirmed floor crossings; SP has 0.

The surface decoder is a shared `Linear(n_hidden, 4)` MLP for all four channels [cp, τx, τy, τz]. There is currently **no mechanism to preferentially steer gradient signal toward the cp (surface pressure) channel**. The H48 TAU-Y-EQUALIZE experiment showed that a single loss-weight flag (`--tau-y-loss-weight 1.0`, reduced from 1.5) produced the largest τz/τx mean deflection in fleet history — every car outside the attractor band. This is the same mechanism, targeted at SP.

**Hypothesis:** Up-weighting the cp channel in the surface MSE loss (`--cp-loss-weight 2.0`) forces the shared `Linear(n_hidden, 4)` surface decoder to route more gradient capacity toward cp reconstruction, potentially breaking the SP floor.

## Implementation

This is a minimal loss-term modification. The complete change to `train.py`:

### 1. Add `cp_loss_weight` to `TrainConfig` (around line 106)

```python
# Before (lines 106-107):
    tau_y_loss_weight: float = 1.0
    tau_z_loss_weight: float = 1.0

# After:
    cp_loss_weight: float = 1.0
    tau_y_loss_weight: float = 1.0
    tau_z_loss_weight: float = 1.0
```

### 2. Update `surface_channel_weights` tensor (lines 753-760)

```python
# Before:
        surface_channel_weights = torch.tensor(
            [1.0, 1.0, config.tau_y_loss_weight, config.tau_z_loss_weight],
            device=device,
            dtype=torch.float32,
        )
        use_channel_weights = bool(
            (config.tau_y_loss_weight != 1.0) or (config.tau_z_loss_weight != 1.0)
        )

# After:
        surface_channel_weights = torch.tensor(
            [config.cp_loss_weight, 1.0, config.tau_y_loss_weight, config.tau_z_loss_weight],
            device=device,
            dtype=torch.float32,
        )
        use_channel_weights = bool(
            (config.cp_loss_weight != 1.0)
            or (config.tau_y_loss_weight != 1.0)
            or (config.tau_z_loss_weight != 1.0)
        )
```

### 3. Update logging (lines 851-854, 884-885, 1069-1071)

```python
# Line 851-854 print statement:
            if use_channel_weights:
                print(
                    f"Surface channel weights: cp={config.cp_loss_weight} tau_x=1.0 "
                    f"tau_y={config.tau_y_loss_weight} tau_z={config.tau_z_loss_weight}"
                )

# Lines 884-885 wandb.summary — add cp:
            wandb.summary["loss/cp_loss_weight"] = config.cp_loss_weight
            wandb.summary["loss/tau_y_loss_weight"] = config.tau_y_loss_weight
            wandb.summary["loss/tau_z_loss_weight"] = config.tau_z_loss_weight

# Lines 1069-1071 train step log — add cp:
                            "train/cp_loss_weight": config.cp_loss_weight,
                            "train/tau_y_loss_weight": config.tau_y_loss_weight,
                            "train/tau_z_loss_weight": config.tau_z_loss_weight,
```

### 4. Add CLI argument parsing

Search for `--tau-y-loss-weight` in the argparse section and add the analogous:

```python
parser.add_argument("--cp-loss-weight", type=float, default=1.0,
                    help="Loss weight for the cp (surface pressure) channel.")
```

And wire it to `TrainConfig` the same way as `tau_y_loss_weight`.

## Experiment Arms

Run **two arms in parallel** to bracket the response:

**Arm A: `--cp-loss-weight 2.0`** (moderate up-weighting, analogous to tau_y precedent)
```bash
CUDA_VISIBLE_DEVICES=0,1,2,3 torchrun --nproc_per_node=4 train.py \
  --cp-loss-weight 2.0 \
  --ema_decay 0.9999 \
  --wandb_group h53-cp-loss-weight \
  --wandb_run_name h53-arm-a-cpw2 \
  [... full standard flags ...]
```

**Arm B: `--cp-loss-weight 3.0`** (stronger signal)
```bash
CUDA_VISIBLE_DEVICES=4,5,6,7 torchrun --nproc_per_node=4 train.py \
  --cp-loss-weight 3.0 \
  --ema_decay 0.9999 \
  --wandb_group h53-cp-loss-weight \
  --wandb_run_name h53-arm-b-cpw3 \
  [... full standard flags ...]
```

Use 4 GPUs per arm (all 8 GPUs in use). Keep all other flags identical to the current best checkpoint recipe.

## Kill Thresholds

Kill if at step 10864 (EP1 boundary):
- `val_abupt > 8.0%` — cp up-weighting destroying WSS too aggressively
- `val_surf_p > 5.0%` — SP getting WORSE despite up-weighting (dead end)

If an arm's val_surf_p is rising at EP1 rather than falling, kill it immediately.

## Research Note

This is a **loss-term modification** — run `@researcher-agent` before implementation to:
1. Understand how cp up-weighting interacts with GradNorm if `--use-gradnorm` is enabled (GradNorm dynamically reweights tasks; explicit channel weights apply inside the surface MSE, GradNorm reweights surface vs volume task split — they act at different levels)
2. Consider whether `--cp-loss-weight` should be co-varied with `--tau-y-loss-weight` or `--tau-z-loss-weight` to maintain WSS relative ratios
3. Note that GradNormBalancer uses task names `("sp", "tau_x", "tau_y", "tau_z", "vp")` — `sp` here is surface pressure, confirming this is the same channel we target

You may adapt the weight values if research reveals a clearly better range; note any deviation in the PR.

## Baseline Metrics (Current Best — PR #972)

| Metric | Baseline | AB-UPT SOTA | Floor (must not breach) |
|--------|----------|-------------|--------------------------|
| val_abupt | 6.126% | — | ≤ 6.126% (merge gate) |
| test_vol_p | 3.643% | 6.08% | ≤ 3.643% |
| **test_SP** | **3.577%** | **3.82%** | **≤ 3.577% (binding gate — 0 crossings Wave 31)** |
| test_WSS_agg | — | 7.29% | — |
| test_τx | — | 5.35% | — |
| test_τy | — | 3.65% | — |
| test_τz | — | 3.63% | — |

**SP context:** Our baseline already beats AB-UPT SOTA (3.577% vs 3.82%). Every Wave 31 experiment regresses test_SP to 3.7–3.9%, breaching the floor. Zero floor crossings in Wave 31 despite 7 on vol_p.

## Monitoring

Track per-epoch in W&B:
- `val/surf_p` — must trend down toward ≤ 3.577%
- `val/abupt` — must stay ≤ 6.126% at convergence
- `val/wall_shear_z` / `val/wall_shear_x` — watch τz/τx ratio for mechanism class signal
- `val/vol_p` — do not regress above 3.643%

Report per-car τz/τx std and mean at each EP gate (same convention as H48).

## Terminal Result Format

```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["<run-id-a>","<run-id-b>"],"primary_metric":{"name":"val_abupt","value":<number>},"test_metric":{"name":"test_surf_p","value":<number>}}
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #1201 because ADVISOR: Closing PR #1201 because the branch is named `dl24-tanjiro/h53-cp-loss-weight` (dl24 advisor prefix convention) but the assigned student is OUR tanjiro under the tay advisor. Our tanjiro student-polling looks for branches under `tanjiro/...` prefix — this branch would not be picked up. Recreating the H53 hypothesis on the correct branch `tanjiro/h53-cp-loss-weight` in a new PR. The hypothesis content is sound and will be preserved verbatim..
